drivers/gpio/CMakeLists.txt

```diff
@@ -7,6 +7,7 @@ zephyr_sources_ifdef(CONFIG_GPIO_DW         gpio_dw.c)
 zephyr_sources_ifdef(CONFIG_GPIO_ESP32      gpio_esp32.c)
 zephyr_sources_ifdef(CONFIG_GPIO_FE310      gpio_fe310.c)
 zephyr_sources_ifdef(CONFIG_GPIO_GECKO      gpio_gecko.c)
+zephyr_sources_ifdef(CONFIG_GPIO_IMX        gpio_imx_gpio.c)
```

Comment:
Do you think is possible to use the same GPIO driver from PR #7113?

Reply:
Yes, should be possible. We should select one or the other.

Reply:
Instead of gpio_imx_gpio.c can we use gpio_imx.c?
This way the filename is more clear.

Reply:
Ok. Note that the FreeRTOS BSP driver is also gpio_imx.c, hopefully it won't cause problems in any toolchains.

Reply:
Extracted this part into a separated PR #7306 so #7113 can rebase to it.

Reply:
@stanislav-poboril,

I had this name in my PR and no issues with the compilation.